### ISSUE TITLE
$transition$.params() includes a "#" key

### ISSUE THREAD

**Issue by tandrewnichols** [angular-ui/ui-router#3744]
I'm trying to transition from `$stateParams` to `$transition$.params()` in an Angular.js (current at 1.4) app. Things are mostly working except that I'm seeing an extra key in all my params.

Here's a relevant snippet from my router:

```javascript
// app/admin/admin-router.js (lines 2-25)
$uiViewScrollProvider.useAnchorScroll();
$urlMatcherFactoryProvider.strictMode(false);

$urlRouterProvider.otherwise('member');

// Base admin area
$stateProvider.state('admin', {
  url: '/',
  templateUrl: 'admin/index.html',
  controller: 'AdminController'
});

// Member search etc.
$stateProvider.state('admin.member', {
  url: 'member',
  templateUrl: 'admin/member/search.html',
  controller: 'MemberSearchController'
})
.state('admin.member.results', {
  url: '/results?subId&email&slug&phone&firstname&lastname&partialEmail&role',
  templateUrl: 'admin/member/search-results.html',
  controller: 'MemberSearchResultsController'
})
```

In the `MemberSearchResultsController`, the parameters are populated by a search form. If I fill one field in and log params inside the controller I get this:

```javascript
{
  #: null,
  subId: 'banana',
  email: undefined,
  phone: undefined,
  firstname: undefined,
  lastname: undefined,
  partialEmail: undefined,
  role: undefined
}
```

which is a problem because I'm currently blindly serializing that to send to a remote API and `null` is not dropped the way `undefined` is, so I end up with something like `http://mydomain.com/myroute?%23=null&subId=banana`. Obviously, I can work around this case, but everywhere I've checked params, they always include this extraneous "#" key. Why is that? Is this something legacy about my app or an issue with ui-router? FWIW, if I log `$transition$.targetState().params()` the `#` key is not present.

I'm also setting

```
  $locationProvider.html5Mode(false);
  $locationProvider.hashPrefix('!');
```

in a different config block in case that's relevant (and I suspect that it is).

This is a (check one box):

- [x] Bug Report
- [ ] Feature Request
- [ ] General Query 

My version of UI-Router is: (type version)

1.0.17

# Bug Report

#### Current Behavior:

I get an extra key in my params: "#"

#### Expected Behavior:

I get only the params I specify in my routing config.

**Comment by tandrewnichols** [comment]
That doesn't seem related to the issue I posted.

**Comment by soelinn** [comment]
You are right that it is not directly related to your reported issue.
However, I am convinced that the issues we are reporting are indirectly related due to the AngularJS version used.

I also noticed the `$stateParams["#"]` value in the `$stateParams` injected into the Angular Controller as you reported **depending on the AngularJS version used**.

I posted my code because it is easier for the ui-router maintainers to test.
